exec/java-exec/src/test/java/org/apache/drill/exec/fn/impl/TestCastFunctions.java

```diff
@@ -78,4 +81,71 @@ public void testToDateForTimeStamp() throws Exception {
         .build()
         .run();
   }
+
+  @Test // DRILL-4970
+  public void testCastNegativeFloatToInt() throws Exception {
+    try {
+      test("create table dfs_test.tmp.table_with_float as\n" +
```

Comment:
The initial bug was caused by changing the input holder when negative values casting, so current test checks the correctness of the fix only.
Added tests to check cast functions with boundary values.